pkg/cache/clusterqueue_test.go

```diff
@@ -98,206 +98,281 @@ func TestFitInCohort(t *testing.T) {
 				{Flavor: "f2", Resource: corev1.ResourceMemory}: 1,
 			},
-			wantFit: false,
-			cq: &ClusterQueueSnapshot{
-				Name: "CQ",
-				Cohort: &CohortSnapshot{
-					Name: "C",
-					RequestableResources: resources.FlavorResourceQuantitiesFlat{
-						{Flavor: "f1", Resource: corev1.ResourceCPU}:    5,
-						{Flavor: "f1", Resource: corev1.ResourceMemory}: 5,
-					}.Unflatten(),
-					Usage: resources.FlavorResourceQuantitiesFlat{
-						{Flavor: "f1", Resource: corev1.ResourceCPU}:    5,
-						{Flavor: "f1", Resource: corev1.ResourceMemory}: 5,
-					}.Unflatten(),
-				},
-				ResourceGroups: nil,
+			wantFit:     false,
+			flavorNames: []string{"f1"},
+			usage: resources.FlavorResourceQuantitiesFlat{
+				{Flavor: "f1", Resource: corev1.ResourceCPU}:    5_000,
+				{Flavor: "f1", Resource: corev1.ResourceMemory}: 5,
+			},
+			clusterQueue: []*kueue.ClusterQueue{
+				utiltesting.
+					MakeClusterQueue("CQ").
+					ResourceGroup(
+						utiltesting.MakeFlavorQuotas("f1").
+							ResourceQuotaWrapper(corev1.ResourceCPU).NominalQuota("5").Append().
+							FlavorQuotas,
+					).
+					ResourceGroup(
+						utiltesting.MakeFlavorQuotas("f1").
+							ResourceQuotaWrapper(corev1.ResourceMemory).NominalQuota("5").Append().
+							FlavorQuotas,
+					).
+					Cohort("C").
+					Obj(),
 			},
 		},
 		"missing resource": {
 			request: resources.FlavorResourceQuantitiesFlat{
-				{Flavor: "f1", Resource: corev1.ResourceCPU}:    1,
+				{Flavor: "f1", Resource: corev1.ResourceCPU}:    1_000,
 				{Flavor: "f1", Resource: corev1.ResourceMemory}: 1,
 			},
-			wantFit: false,
-			cq: &ClusterQueueSnapshot{
-				Name: "CQ",
-				Cohort: &CohortSnapshot{
-					Name: "C",
-					RequestableResources: resources.FlavorResourceQuantitiesFlat{
-						{Flavor: "f1", Resource: corev1.ResourceCPU}: 5,
-					}.Unflatten(),
-					Usage: resources.FlavorResourceQuantitiesFlat{
-						{Flavor: "f1", Resource: corev1.ResourceCPU}: 3,
-					}.Unflatten(),
-				},
-				ResourceGroups: nil,
+			wantFit:     false,
+			flavorNames: []string{"f1"},
+			usage: resources.FlavorResourceQuantitiesFlat{
+				{Flavor: "f1", Resource: corev1.ResourceCPU}: 3_000,
+			},
+			clusterQueue: []*kueue.ClusterQueue{
+				utiltesting.
+					MakeClusterQueue("CQ").
+					ResourceGroup(
+						utiltesting.MakeFlavorQuotas("f1").
+							ResourceQuotaWrapper(corev1.ResourceCPU).NominalQuota("4").Append().
+							FlavorQuotas,
+					).
+					Cohort("C").
+					Obj(),
 			},
 		},
 		"lendingLimit enabled can't fit": {
 			request: resources.FlavorResourceQuantitiesFlat{
-				{Flavor: "f1", Resource: corev1.ResourceCPU}: 3,
+				{Flavor: "f1", Resource: corev1.ResourceCPU}: 3_000,
 			},
-			wantFit: false,
-			cq: &ClusterQueueSnapshot{
-				Name: "CQ-A",
-				Cohort: &CohortSnapshot{
-					Name: "C",
-					RequestableResources: resources.FlavorResourceQuantitiesFlat{
-						{Flavor: "f1", Resource:
-						// CQ-A has 2 nominal cpu, CQ-B has 3 nominal cpu and 2 lendingLimit,
```

Comment:
here and in the next test:

is it worth encoding these comments in the test setup, now that we have the capability of specifying lending limit?

Reply:
I don't think it's worth encoding comments.
I believe that by looking at the test data, one can understand what the test is about.

Reply:
In this case, it is worth changing. These two tests no longer match the intent of the original test.

The original test was pretending to have a lending limit by configuring the internal state to what it would be had we specified a lending limit. Now, we can use lending limit in the  kueue.ClusterQueue type to express this concept directly.

Please update these tests to match the nominal quotas/lending limits in the comments. (to simplify, since we hardcode `CQ` below when retrieving from snapshot, `CQ-A` can be called `CQ`, and `CQ-B` stays the same